doc/dash-acl-tags/dash-acl-tags.md

```diff
@@ -0,0 +1,277 @@
+- If tags are specified in the configuration
+    - For each configured tag
+        - Check if the tags exists in `DASH_PREFIX_TAG_TABLE`. If the tag doesn't exist return status "retry" to postpone rule creation.
+    - If all tags are available
+      - Expand all tags to a list of prefixes
+      - Create ACL rule
+
+**Remove ACL rule**
+
+If the operation is to remove the existing ACL rule from DASH_ACL_RULE_TABLE DashAclOrch will do the following
+- Remove rule
+
+**Update ACL rule**
+
+If the operation is to update the existing ACL rule in DASH_ACL_RULE_TABLE DashAclOrch will do the following
+- if the ACL group to which the ACL rule belongs is bound to the ENI
+  - Create a new ACL group
+  - For each ACL rule from the original ACL group
+    - Copy original ACL rule attributes
+    - Update attributes if required
+    - Create a new ACL rule and attach it to the new ACL group
+- Else
+  - For each changed attribute
+    - Update attribute configuration
+
+# SAI API
+
+Stage 1 implementation does not require SAI API.
+
+# Configuration and management
+
+## CLI/YANG model Enhancements
+
+No CLI/YANG enhancements required.
+
+## Config DB Enhancements
+
+No Config DB Enhancements required
+
+# Warmboot and Fastboot Design Impact
+DPU SONiC doesn't support warm/fast boot
+
+# Restrictions/Limitations
+
+# Testing Requirements/Design
+
+## Unit Test cases
+
+The existing sonic-swss/tests/test_dash_acl.py unit tests shall be extended with the following test cases
+| Test case synopsis                                       | Expected results                                        |
+| -------------------------------------------------------- | ------------------------------------------------------- |
+| Create ACL rule with a single source and single destination tags | Verify that ACL rule with correct attributes is created |
+| Create ACL rule with multiple source and multiple destination tags | Verify that ACL rule with correct attributes is created |
+| Create ACL rule with multiple source and multiple destination tags and list of source and destination prefixes | Verify that ACL rule with correct attributes is created |
+| Update ACL tag used by ACL rule from ACL group not bound to ENI | Verify that ACL rule with is updated with correct attributes |
+|Update ACL tag used by ACL rule from ACL group bound to the ENI | Verify the following <br />- new ACL group is created <br /> - all ACL rules are copied to the new group <br /> - appropriate rule is updated <br /> - new ACL group is bound to the ENI <br /> - old ACL rules are removed <br /> - old ACL group is removed |
+|Create tag with missing mandatory fields|Verify that error message is sent to syslog|
+|Create tag with incorrect data | Verify that error message is sent to syslog|
+|Create ACL rule with not existing tag |Verify that rule is not created|
+|Create ACL rule with not existing tag than create a tag |Verify that rule is created after the tag is added|
+
```

Comment:
Can we add a test case to confirm delayed creation logic? First lets create a rule with non existing tag and then add the tag later.
Similarly lets remove a tag first and then remove the associated rule later. We need to verify both tag and rule is deleted in this scenario.

Reply:
Sure, I will update the test cases